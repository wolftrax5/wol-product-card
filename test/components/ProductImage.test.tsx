import React from 'react';
import render from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/products';

describe('Product Image', () => {
  test('debe de mostrar la url Personalizado', () => {
    const wrapper = render.create(
      <ProductImage
        img="https://avatars.githubusercontent.com/u/5388080?v=4"
        className="image-class"
      />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('debe de mostrar el compoenete correctamente con la imagen Personalizado', () => {
    const wrapper = render.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
