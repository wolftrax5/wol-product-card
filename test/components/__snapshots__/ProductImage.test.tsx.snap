// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Image debe de mostrar el compoenete correctamente con la imagen Personalizado 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`Product Image debe de mostrar la url Personalizado 1`] = `
<img
  alt="Product"
  className="productImg image-class"
  src="https://avatars.githubusercontent.com/u/5388080?v=4"
/>
`;
